### ISSUE TITLE
False positive for convenience_type with Actors

### ISSUE THREAD

**Issue by StarLard** [realm/SwiftLint#3770]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint incorrectly fires a `convenience_type` violation when using `Actor` types. Enums are not a co-equal replacement for Actors. This is a common use case for global actors.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
error: Convenience Type Violation: Types used for hosting only static members should be implemented as a caseless enum to avoid instantiation. (convenience_type)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.45.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. NA
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 13.2 beta 2
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
@globalActor actor MyActor {
  static let shared = MyActor()
}
```

**Comment by jpsim** [comment]
Fixed in https://github.com/realm/SwiftLint/pull/3812.
